components/waitwhile/actions/add-customer-note-entry/add-customer-note-entry.ts

```diff
@@ -0,0 +1,52 @@
+import waitwhile from "../../app/waitwhile.app";
+import { defineAction } from "@pipedream/types";
+
+export default defineAction({
+  name: "Add Customer Note Entry",
+  version: "0.0.1",
+  key: "add-customer-note-entry",
+  description: "Add a customer note entry",
+  props: {
+    waitwhile,
+    customerId: {
+      label: "CustomerID",
+      type: "string",
+      description: "Identifier of customer, automatically derived from visitor contact information if not provided."
+    },
+    content: {
+      label: "Content",
+      type: "string",
+      optional: true,
+      description: "Optional notes"
+    },
+    locationId: {
+      label: "Location ID",
+      type: "string",
+      optional: true,
+      description: "Identifier of location"
+    },
+    visitId: {
+      label: "Visit ID",
+      type: "string",
+      optional: true,
+      description: "Identifier of visit"
+    }
```

Comment:
These props (`locationId` and `visitId`) could also be placed in the app file and set up to use [async options](https://pipedream.com/docs/components/migrating/#advanced-using-async-options).